File: LC605.py
class Solution:
    def canPlaceFlowers(self, a: List[int], n: int) -> bool:
        l = len(a)

        for i in range(l):
            if not a[i]:
                left = (not i) or (not a[i-1])
                right = (i == l - 1) or (not a[i+1])

                if left and right:
                    a[i] = 1
                    n -= 1
                    if n <= 0:
                        return True
        
        return n <= 0

File: LC605.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& a, int n) {
        int i = 0, l = a.size();
        for(int i = 0; i < l; i++){
            if(!a[i]){
                bool left = (i == 0) || (a[i-1] == 0);
                bool right = (i == l-1) || (a[i+1] == 0);

                if(left and right){
                    a[i] = 1;
                    n--;
                    if(n <= 0)
                        return true;
                }
            }
        }

        return n <= 0;
    }
};
